arch/arm/boot/dts/overlays/revpi-connect4-overlay.dts

```diff
@@ -0,0 +1,294 @@
+					gpios = <&expander_power 0 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_red";
+				};
+				a3_green {
+					label = "a3:green:status";
+					gpios = <&expander_power 1 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_green";
+				};
+				a3_blue {
+					label = "a3:blue:status";
+					gpios = <&expander_power 2 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_blue";
+				};
+				a4_red {
+					label = "a4:red:status";
+					gpios = <&expander_power 8 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_red";
+				};
+				a4_green {
+					label = "a4:green:status";
+					gpios = <&expander_power 9 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_green";
+				};
+				a4_blue {
+					label = "a4:blue:status";
+					gpios = <&expander_power 10 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_blue";
+				};
+				a5_red {
+					label = "a5:red:status";
+					gpios = <&expander_power 11 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_red";
+				};
+				a5_green {
+					label = "a5:green:status";
+					gpios = <&expander_power 12 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_green";
+				};
+				a5_blue {
+					label = "a5:blue:status";
+					gpios = <&expander_power 13 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_blue";
+				};
+			};
+		};
+	};
+
+	/* PINMUX */
+	fragment@1 {
+		target = <&gpio>;
+		__overlay__ {
+			pb_uart_pins: pb_uart_pins {
+				/* TX RX TX_EN */
+				brcm,pins     = <4 5 7>;
+				brcm,function = <BCM2835_FSEL_ALT4>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			digital_io_pins: digital_io_pins {
+				/* digital IN digital OUT */
+				brcm,pins     = <6 16>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN
+				                 BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF
+				                 BCM2835_PUD_OFF>;
+			};
+			exp_core_pins: exp_core_pins {
+				brcm,pins     = <8>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			exp_power_pins: exp_power_pins {
+				brcm,pins     = <9>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			conn-i2c1_pins: conn-i2c1_pins {
+				/* SDA SCL */
+				brcm,pins     = <2 3>;
+				brcm,function = <BCM2835_FSEL_ALT0>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			rs485_pins: rs485_pins {
+				/* TX RX TX_EN */
+				brcm,pins     = <12 13 15>;
+				brcm,function = <BCM2835_FSEL_ALT4
+				                 BCM2835_FSEL_ALT4
+				                 BCM2835_FSEL_ALT4>;
+				brcm,pull     = <BCM2835_PUD_OFF
+				                 BCM2835_PUD_OFF
+				                 BCM2835_PUD_DOWN>;
+			};
+			spi1_cs_pins: spi1_cs_pins {
+				/* tpm (SPI1_CE0_N) */
+				brcm,pins     = <18>;
+				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_DOWN>;
+			};
+			spi1_pins: spi1_pins {
+				/* MISO MOSI SCLK */
+				brcm,pins     = <19 20 21>;
+				brcm,function = <BCM2835_FSEL_ALT4>;
+				brcm,pull     = <BCM2835_PUD_DOWN>;
```

Comment:
No pull-down here, the MOSI and the SCLK are driven by the SPI controller. And the MISO is driven by the device.